server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/PayloadCodecProvider.java

```diff
@@ -0,0 +1,58 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.objectstorage;
+
+import static org.apache.james.modules.objectstorage.ObjectStorageConfiguration.OBJECTSTORAGE_CONFIGURATION_NAME;
+import static org.apache.james.modules.objectstorage.ObjectStorageConfiguration.OBJECTSTORAGE_PAYLOAD_CODEC;
+
+import java.io.FileNotFoundException;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.james.blob.objectstorage.PayloadCodec;
+import org.apache.james.utils.PropertiesProvider;
+
+import com.google.common.base.Preconditions;
+
+public class PayloadCodecProvider implements Provider<PayloadCodec> {
+    private final Configuration configuration;
+
+    @Inject
+    public PayloadCodecProvider(PropertiesProvider propertiesProvider)
+        throws ConfigurationException {
+        try {
+            this.configuration =
+                propertiesProvider.getConfiguration(OBJECTSTORAGE_CONFIGURATION_NAME);
+        } catch (FileNotFoundException e) {
+            throw new ConfigurationException(OBJECTSTORAGE_CONFIGURATION_NAME + "configuration was not found");
+        }
+    }
+
+    @Override
+    public PayloadCodec get() {
+        String endpointStr = configuration.getString(OBJECTSTORAGE_PAYLOAD_CODEC, null);
+        Preconditions.checkArgument(endpointStr != null,
+            OBJECTSTORAGE_PAYLOAD_CODEC + "is a mandatory configuration value");
+        return PayloadCodecs.valueOf(endpointStr).codec(configuration);
```

Comment:
We tend to not use `enum::valueOf` much the use is interesting but I wonder if the exception in case I do not provide the exact valid string might be cryptic.

What we generally do is:
 - Assign a name field to each enum entry
 - Then write our own  static factory method for doing the validation

Thus, we can specify some sanitizing logic (spaces, case) some different values, as well as specify thrown exception.